app/service/CasesService.scala

```diff
@@ -20,14 +20,27 @@ import connectors.BindingTariffClassificationConnector
 import javax.inject.{Inject, Singleton}
 import models.{Case, NewCaseRequest}
 import uk.gov.hmrc.http.HeaderCarrier
+import scala.concurrent.ExecutionContext.Implicits.global
 
 import scala.concurrent.Future
 
 @Singleton
-class CasesService @Inject()(connector: BindingTariffClassificationConnector){
+class CasesService @Inject()(connector: BindingTariffClassificationConnector) {
 
   def create(c: NewCaseRequest)(implicit hc: HeaderCarrier): Future[Case] = {
     connector.createCase(c)
   }
 
+  def getCaseForUser(userEori: String, reference: String)(implicit hc: HeaderCarrier): Future[Option[Case]] = {
+
+    connector.findCase(reference).map {
```

Comment:
I believe this code could be more concise

Reply:
I think a ‘filter’ would work instead of a map

Reply:
I added your suggested method to the Case and have attempted to implement the filter - 
```connector.findCase(reference).filter(maybeCase => maybeCase.exists(_.hasEoriNumber(userEori)))```

This works until the case is filtered out (because it belongs to another Eori) and then I get `java.util.NoSuchElementException: Future.filter predicate is not satisfied`

Reply:
I was thinking it would just become 
`connector.findCase(reference).filter(_.filter(_.hasEori(userEori)))`

Reply:
I take it back, I think you need to keep the `map` but you can use a `filter` inside it

Reply:
That form works (the first one).  Thanks